### PR TITLE
Ssh keys fix

### PR DESCRIPTION
**Pull request by lab-lamz4**
Changed work with ssh-keys for jenkins slave

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -115,14 +115,40 @@ Requirements
...
+```yaml
+- name: Install and Configure Jenkins
+  hosts: master
+  vars:
+    jenkins2_ssh_keys_generate: true
+  roles:
+    - role: lean_delivery.jenkins
+
+- name: Install Jenkins to nodes
+  hosts: j-nodes
```
Comment (ViachaslauKabak) [inline README.md]: roles?

File: tasks/configure_ssh_keys.yml [hunk 2]
```diff
@@ -1,103 +1,64 @@
...
+    register: public_key_stat
 
-- name: Ensure that openssh *public* key doesn't exist
-  stat:
-    path: "{{ jenkins2_home_directory }}/.ssh/\
-    {{ jenkins2_ssh_keys_private_keyname }}.pub"
-  become: true
-  register: public_key_stat
+  - name: "Generate a pair of keys"
+    command: >-
```
Comment (ViachaslauKabak) [inline tasks/configure_ssh_keys.yml]: use modules please

File: tasks/configure_ssh_keys.yml [hunk 3]
```diff
@@ -1,103 +1,64 @@
...
+      {{ 'ssh-keygen
+      -q
+      -b ' + jenkins2_ssh_keys_size|string + '
+      -t rsa
+      -f ' + jenkins2_home_directory + '/.ssh/' +
+             jenkins2_ssh_keys_private_keyname +
+      ' -C "Jenkins"
+      -N ""' }}
+    changed_when: false
+    when: not private_key_stat.stat.exists|bool
```
Comment (ViachaslauKabak) [inline tasks/configure_ssh_keys.yml]: use spaces among |

File: defaults/main.yml [hunk 4]
```diff
@@ -52,8 +52,8 @@ jenkins2_local_cert_file_name: '{{ ansible_hostname }}.ca-cert.pem'
 # SSH_KEYS generation properties
 jenkins2_ssh_keys_generate: false
 jenkins2_ssh_keys_private_keyname: "id_rsa"
-jenkins2_ssh_keys_size: 4096
-jenkins2_ssh_keys_inv_slave_groupname: "jenkins_ssh_slaves"
+jenkins2_ssh_keys_size: 2048
+jenkins2_ssh_slave_key_fetch: '/tmp/slave.pub'
 
 jenkins2_ssh_keys_slave_group: "{{ groups[jenkins2_ssh_keys_inv_slave_groupname]|default([''])}}"
```
Comment (ViachaslauKabak) [inline defaults/main.yml]: jenkins2_ssh_keys_inv_slave_groupname was removed from readme and defaults

File: README.md (hunk 1 above)
Comment (lab-lamz4) [reply README.md]: Added role: lean_delivery.jenkins_slave

File: tasks/configure_ssh_keys.yml (hunk 3 above)
Comment (lab-lamz4) [reply tasks/configure_ssh_keys.yml]: Fixed

File: defaults/main.yml (hunk 4 above)
Comment (lab-lamz4) [reply defaults/main.yml]: Fixed

File: tasks/configure_ssh_keys.yml (hunk 2 above)
Comment (lab-lamz4) [reply tasks/configure_ssh_keys.yml]: Module openssl_privatekey don't properly work for jenkins credential. In ansible 2.8 have a module named openssh_keypair, but we have an issue with work molecule and ansible 2.8. So, for now, command module its workaround these problems.

File: tasks/configure_ssh_keys.yml [hunk 5]
```diff
@@ -1,103 +1,63 @@
...
+      -N ""' }}
+    when: not private_key_stat.stat.exists | bool
+
+  - name: "Set corect folders owners/permissions"
+    file:
+      path: "{{ item.dst }}"
+      owner: "{{ jenkins2_user | default('jenkins') }}"
+      group: "{{ jenkins2_user | default('jenkins') }}"
+      mode: "{{ item.mode }}"
+    with_items:
```
[the remark this answers is not in the document]
Comment (lab-lamz4) [reply tasks/configure_ssh_keys.yml]: Done

File: tasks/configure_ssh_keys.yml [hunk 6]
```diff
@@ -1,103 +1,63 @@
...
+      - {dst: "{{ jenkins2_home_directory }}/.ssh", mode: "0700"}
```
[the remark this answers is not in the document]
Comment (lab-lamz4) [reply tasks/configure_ssh_keys.yml]: Done

**Pull request closed without merging**